examples/examples/compose_extrinsic.rs

```diff
@@ -0,0 +1,140 @@
+//! - Compose an extrinsic without asking the node for nonce and without knowing the metadata
+
+use codec::Compact;
+use kitchensink_runtime::{BalancesCall, RuntimeCall};
+use sp_keyring::AccountKeyring;
+use sp_runtime::{generic::Era, MultiAddress};
+use substrate_api_client::{
+	ac_compose_macros::{compose_call, compose_extrinsic_offline},
+	ac_primitives::{
+		config::Config, AssetRuntimeConfig, AssetTip, ExtrinsicParams, ExtrinsicSigner,
+		GenericAdditionalParams, SignExtrinsic,
+	},
+	rpc::JsonrpseeClient,
+	Api, GetChainInfo, SubmitAndWatch, XtStatus,
+};
+
+type AssetExtrinsicSigner = <AssetRuntimeConfig as Config>::ExtrinsicSigner;
+type AccountId = <AssetRuntimeConfig as Config>::AccountId;
+type ExtrinsicAddressOf<Signer> = <Signer as SignExtrinsic<AccountId>>::ExtrinsicAddress;
+
+type Hash = <AssetRuntimeConfig as Config>::Hash;
+/// Get the balance type from your node runtime and adapt it if necessary.
+type Balance = <AssetRuntimeConfig as Config>::Balance;
+/// We need AssetTip here, because the kitchensink runtime uses the asset pallet. Change to PlainTip if your node uses the balance pallet only.
+type AdditionalParams = GenericAdditionalParams<AssetTip<Balance>, Hash>;
+
+type Address = <AssetRuntimeConfig as Config>::Address;
+
+// To test this example with CI we run it against the Substrate kitchensink node, which uses the asset pallet.
+// Therefore, we need to use the `AssetRuntimeConfig` in this example.
+// ! However, most Substrate runtimes do not use the asset pallet at all. So if you run an example against your own node
+// you most likely should use `DefaultRuntimeConfig` instead.
+
+#[tokio::main]
+async fn main() {
+	env_logger::init();
+
+	// Initialize api and set the signer (sender) that is used to sign the extrinsics.
+	let signer = AccountKeyring::Alice.pair();
+	let client = JsonrpseeClient::with_default_url().unwrap();
+
+	let mut api = Api::<AssetRuntimeConfig, _>::new(client).unwrap();
+	let extrinsic_signer = ExtrinsicSigner::<AssetRuntimeConfig>::new(signer);
+	// Signer is needed to set the nonce and sign the extrinsic.
+	api.set_signer(extrinsic_signer.clone());
+
+	let recipient: Address = MultiAddress::Id(AccountKeyring::Bob.to_account_id());
+
+	// Get the last finalized header to retrieve information for Era for mortal transactions (online).
+	let last_finalized_header_hash = api.get_finalized_head().unwrap().unwrap();
+	let header = api.get_header(Some(last_finalized_header_hash)).unwrap().unwrap();
+	let period = 5;
+
+	// Construct extrinsic without using Api (no_std).
+	let additional_extrinsic_params: AdditionalParams = GenericAdditionalParams::new()
+		.era(Era::mortal(period, header.number.into()), last_finalized_header_hash)
+		.tip(0);
+
+	println!("Compose extrinsic in no_std environment (No Api instance)");
+	{
+		let signer_nonce = api.get_nonce().unwrap();
+		println!("[+] Alice's Account Nonce is {}", signer_nonce);
```

Comment:
Would it be possible to move the api part outside of this environment? That would make it clearer, that really no api client is needed (given the information is availbale).
E.g.
```rust
let spec_version = api.runtime_version().spec_version;
let transaction_version = api.runtime_version().transaction_version;
let genesis_hash = api.genesis_hash
..
let mut xt = ...;
{
        let extrinsic_params = <AssetRuntimeConfig as Config>::ExtrinsicParams::new(
			spec_version,
			transaction_version,
			signer_nonce,
			genesis_hash,
			additional_extrinsic_params,
		);

		let recipients_extrinsic_address: ExtrinsicAddressOf<AssetExtrinsicSigner> =
			recipient.clone().into();

		let call = compose_call!(
			metadata,
			"Balances",
			"transfer_allow_death",
			recipients_extrinsic_address,
			Compact(4u32)
		);
		xt = compose_extrinsic_offline!(extrinsic_signer, call, extrinsic_params);
}
// Test created extrinsic via api call to the node (online)
let hash = api
			.submit_and_watch_extrinsic_until(xt, XtStatus::InBlock)
			.unwrap()
			.block_hash
			.unwrap();

Reply:
Same for the api (offline) part - move the online part outside? 

Does that make sense?